### ISSUE TITLE
Empty arrays detected as cycle

### ISSUE THREAD

**Issue by TheBlue-1** [Faleij/json-stream-stringify#34]
```
{
  raw: { reference: '/items' },
  partiallyEvaluated: { resolved: [] },
  evaluated: []
}
```
results in the error `Error: Converting circular structure to JSON`

If cycle is set to true it will create an result like
`{"raw":{"reference":"/items"},"partiallyEvaluated":{"resolved":[]},"evaluated":{"$ref":"$[\"input\"][\"partiallyEvaluated\"][\"resolved\"]"}}`

It only occurs if the same array is used twice.

Full code example:
```
const { JsonStreamStringify } = require("json-stream-stringify");

const array = [];

const val = {
  raw: { reference: "/items" },
  partiallyEvaluated: { resolved: array },
  evaluated: array,
};

const stream = new JsonStreamStringify(val);

stream.on("data", (chunk) => {
  console.log("data", chunk);
});

stream.on("error", (err) => {
  console.log("err", err);
});

stream.on("close", () => {
  console.log("close");
});

```

**Comment by Faleij** [comment]
Thanks for reporting, will investigate soon.

**Comment by Faleij** [comment]
This is working as it expected and as [Douglas Crockfords decycle](https://github.com/douglascrockford/JSON-js/blob/master/cycle.js) method works. You have a reference to the exact same array instance and thus it gets decycled. If you had `{ a: [], b: [] } ` neither a or b would get decycled because they are not the same array instance.

**Comment by TheBlue-1** [comment]
Yeah i agree that decycling makes sense and i wouldn't change that. (Also didn't in the pr). 
The problem is that it throws the error `Converting circular structure to JSON` if cycle is not set to true, since it should just create a json out of it.

**Comment by TheBlue-1** [comment]
If you run the code example i added to my pr also an error `Converting circular structure to JSON`. But one would expect to receive a json like `{"raw":{"reference":"/items"},"partiallyEvaluated":{"resolved":[]},"evaluated":[]}`

**Comment by Faleij** [comment]
Sorry, I didn't read the issue properly at first, my bad. It is now fixed in v3.1.1 which should be up soon.

**Comment by TheBlue-1** [comment]
Thank you for the quick support and fix
